tests/unit_tests/test_apply_dimension.py

```diff
@@ -0,0 +1,186 @@
+import json
+from queue import Empty
+
+import pytest
+
+from tests.utils import (
+    load_process_code,
+    load_datacube_code,
+    run_process,
+    run_input_validation,
+)
+
+
+@pytest.fixture
+def apply_dimension_process_code():
+    return load_process_code("apply_dimension")
+
+
+@pytest.mark.parametrize(
+    "example_input,expected_result",
+    [
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => data",
+                "dimension": "temporal",
+            },
+            [1, 2, 3, 4, 5, 6],
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => data.map(el => el * 3)",
+                "dimension": "temporal",
+            },
+            [3, 6, 9, 12, 15, 18],
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => {const sum = data.reduce((a, b) => a + b, 0); return data.map(el => el * sum)}",
+                "dimension": "temporal",
+            },
+            [9, 24, 27, 48, 45, 72]
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => data",
+                "dimension": "bands",
+            },
+            [1, 2, 3, 4, 5, 6],
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => data.map(el => el * 3)",
+                "dimension": "bands",
+            },
+            [3, 6, 9, 12, 15, 18],
+        ),
+        (
+            {
+                "data": [
+                    {"B01": 1, "B02": 2},
+                    {"B01": 3, "B02": 4},
+                    {"B01": 5, "B02": 6},
+                ],
+                "process": "({data}) => {const sum = data.reduce((a, b) => a + b, 0); return data.map(el => el * sum)}",
+                "dimension": "bands",
+            },
+            [3, 6, 21, 28, 55, 66]
+        ),
+    ],
+)
+def test_apply_dimension(apply_dimension_process_code, example_input, expected_result):
```

Comment:
Minor: I personally prefer having separate input variables instead of passing one dict and then having to get the values from it. I think it makes it cleaner. fftr.

Reply:
If referring to `example_input` then I think I would agree. But since every other test we have uses this same approach I would rather pick consistency in this case.